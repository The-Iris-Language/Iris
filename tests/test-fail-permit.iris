.. Test calling a permitted function in while loop
class Animal (Main) {
    public: 
        void ignore() {
            Olympus.println("...");
        }

    permit:
        string s;
        void noise() {
            Olympus.println("Noise");
        }
}

class Duck of Animal () {
    permit:
        void noise() {
            Olympus.println("Quack!");
        }
}

class Main () {
    public:
        int univ main() {
            Duck d = new Duck();
            int count = 0;
            if (count == 0) {
                d.noise();
            }
            
            return 0;
        }
}