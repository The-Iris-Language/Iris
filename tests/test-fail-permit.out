RUNTIME ERROR: Class Main does not have access!
